app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/ProfileDetailsFragmentTest.kt

```diff
@@ -55,25 +81,279 @@ class ProfileDetailsFragmentTest {
+            }
+        }
+
+        with(launchFragmentInContainer(bundle) { ProfileDetailsFragment(testRegistry) }) {
+            onFragment {
+                val connectedUserViewModel:
+                        ConnectedUserViewModel by it.viewModels(ownerProducer = { it.requireActivity() })
+                connectedUserViewModel.currentUser.value = user
+
+                // initially check that no profile picture is associated with this user
+                assertThat(connectedUserViewModel.currentUserProfilePicture.value, `is`(nullValue()))
+            }
+
+            // click on button to edit profile
+            onView(withId(R.id.fragment_profile_details_edit_profile))
+                .check(matches(isDisplayed()))
+                .perform(click())
+
+            // take picture
+            onView(withId(R.id.fragment_profile_details_take_picture))
+                .check(matches(isDisplayed()))
+                .perform(click())
+
+            // check that the user has now its picture stored on the db
+
+            onFragment {
+                val connectedUserViewModel:
+                        ConnectedUserViewModel by it.viewModels(ownerProducer = { it.requireActivity() })
+                connectedUserViewModel.currentUser.value = user
+
+                // check that the value is now the taken picture
+                assertThat(connectedUserViewModel.currentUserProfilePicture.value, `is`(sameInstance(fakePicture)))
+            }
+            close()
+        }
+    }
+
+    @Test
+    fun correctInfoDisplayedExistingUserDifferentFromAuthenticatedOne() {
+        val fakeAuthUser = User("1", 1, "Test")
+        val existingUser = User("VaRgQioAuiGtfDlv5uNuosNsACCJ",  0, description = "That's the awesome description of test user 1", displayName = "User Test 1", email = "user1@email.ch", phone = "+41245285397", profilePictureMetadata = PhotoMetadata("VaRgQioAuiGtfDlv5uNuosNsACCJ_profile_picture"))
+
+        val bundle = Bundle().apply { putParcelable(ARG_USER, existingUser) }
+        val frag = launchFragmentInContainer<ProfileDetailsFragment>(bundle)
+
+        frag.onFragment {
+            val connectedUserViewModel:
+                    ConnectedUserViewModel by it.viewModels(ownerProducer = { it.requireActivity() })
+            connectedUserViewModel.currentUser.value = fakeAuthUser
+            connectedUserViewModel.currentUserProfilePicture.value = null
+        }
+
+        onView(withId(R.id.fragment_profile_details_profile_name))
+            .check(matches(withText("User Test 1")))
+
+        onView(withId(R.id.fragment_profile_details_profile_description))
+            .check(matches(withText("That's the awesome description of test user 1")))
+
+        onView(withId(R.id.fragment_profile_details_profile_email))
+            .check(matches(withText("user1@email.ch")))
+
+        onView(withId(R.id.fragment_profile_details_profile_phone))
+            .check(matches(withText("+41245285397")))
```

Comment:
please tell me this isn't a real phone number 😂

Reply:
I didn't try to call it, but I typed it randomly so hopefuly it's not a real one 😂